### ISSUE TITLE
Segfault when running Lua example on macOS 10.14.1

### ISSUE THREAD

**Issue by samdoiron** [wasmerio/wasmer#216]
When running the `lua.wasm` example on macOS `10.14.1` the program segfaults when you attempt to define a function.

Reproduction:
```bash
examples :: λ → wasmer run lua.wasm
Lua 5.4.0  Copyright (C) 1994-2018 Lua.org, PUC-Rio
> function foo
"Runtime(Unknown { msg: \"trap at 0x0 - segmentation violation\" })"
```

**Comment by lachlansneff** [comment]
Hi!

Yeah, we've also run into this issue recently. Lua uses `setjmp` and `longjmp` internally to approximate exceptions and it's actually impossible for us to safely emulate those two functions until we support wasm exceptions.

**Comment by syrusakbary** [comment]
We actually found a solution imitating the error handling that emscripten does in js (thanks to #338).

Lua will be running properly once #334 is merged

**Comment by syrusakbary** [comment]
This is now fixed in wasmer 0.3.0. Closing the issue
